### PR TITLE
Support terrain meshes in the RayTracingFeatureProcessor

### PR DESCRIPTION
**Pull request by dmcdiarmid-ly**
Added terrain mesh to the ray tracing scene.

Tested AtomSampleViewer DX12/Vulkan
Tested Editor DX12/Vulkan

Signed-off-by: dmcdiarmid-ly <63674186+dmcdiarmid-ly@users.noreply.github.com>

### CODE REVIEW

File: Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingFeatureProcessor.cpp [hunk 1]
```diff
@@ -77,30 +77,29 @@ namespace AZ
...
             {
-                AZ_Assert(false, "SetMesh called on an existing Mesh objectId, call RemoveMesh first");
+                AZ_Assert(false, "AddMesh called on an existing Mesh, call RemoveMesh first");
                 return;
             }
 
             // add the mesh
-            m_meshes.insert(AZStd::make_pair(objectIndex, Mesh{ assetId }));
-            Mesh& mesh = m_meshes[objectIndex];
+            m_meshes.insert(AZStd::make_pair(uuid, Mesh{ assetId }));
```
Comment (jeremyong-az) [inline Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingFeatureProcessor.cpp:101]: Nit: I think `insert` returns an iterator/inserted-state pair you can use to retrieve the mesh you just added to avoid the extra lookup on the next line.

**Review by jeremyong-az — APPROVED** [review]

**Review by antonmic — APPROVED** [review]

**Review by invertednormal — APPROVED** [review]
Good first step! - just a few suggestions.

File: Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp [hunk 2]
```diff
@@ -1034,4 +1093,37 @@ namespace Terrain
...
+        desc.m_elementFormat = elementFormat;
+
+        if (name != nullptr)
+        {
+            desc.m_bufferName = name;
+        }
+
+        return AZ::RPI::BufferSystemInterface::Get()->CreateBufferFromCommonPool(desc);
+    }
+
```
[the remark this answers is not in the document]
Comment (dmcdiarmid-ly) [reply Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp:1128]: I originally started with that, but it gets a little hard to follow due to the way that the RPI::CommonBufferDescriptor leverages the Unknown format to make decisions about the size.

It's much more clear to have it separated, and I don't think maintenance will be a problem since it's just building a small descriptor.
